gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-mqtt5/src/main/java/io/gravitee/plugin/endpoint/mqtt5/Mqtt5EndpointConnector.java

```diff
@@ -0,0 +1,255 @@
+                        .response()
+                        .messages(
+                            Flowable.defer(
+                                () -> {
+                                    Mqtt5RxClient rxClient = Mqtt5Client
+                                        .builder()
+                                        .identifier(getIdentifier(ctx))
+                                        .serverHost(configuration.getServerHost())
+                                        .serverPort(configuration.getServerPort())
+                                        .automaticReconnectWithDefaultConfig()
+                                        .buildRx();
+
+                                    return RxJavaBridge.toV3Flowable(
+                                        rxClient
+                                            .connect()
+                                            .flatMapPublisher(
+                                                mqttConnAck ->
+                                                    rxClient
+                                                        .subscribePublishesWith()
+                                                        .topicFilter(getTopic(ctx))
+                                                        .qos(Mqtt5Publish.DEFAULT_QOS)
+                                                        .applySubscribe()
+                                                        .map(
+                                                            mqttPublish -> {
+                                                                DefaultMessage.DefaultMessageBuilder messageBuilder = DefaultMessage.builder();
+
+                                                                if (mqttPublish.getPayload().isPresent()) {
+                                                                    messageBuilder.content(Buffer.buffer(mqttPublish.getPayloadAsBytes()));
+                                                                }
+
+                                                                messageBuilder.metadata(
+                                                                    Maps
+                                                                        .<String, Object>builder()
+                                                                        .put("topic", mqttPublish.getTopic())
+                                                                        .put("type", mqttPublish.getType().name())
+                                                                        .put("qos", mqttPublish.getQos().getCode())
+                                                                        .put("retain", mqttPublish.isRetain())
+                                                                        .put("contentType", mqttPublish.getContentType())
+                                                                        .put(
+                                                                            "messageExpiryInterval",
```

Comment:
I wasn't able to properly test the `messageExpiryInterval` feature.
Maybe it is too soon for this PR and we will have to spend more time on it later. From what I saw here (https://www.hivemq.com/blog/mqtt5-essentials-part4-session-and-message-expiry/), `messageExpiryInterval` can be used to implement a kind of retention mechanism. Depending whether the retain flag is set or not, the behavior is not the same. Also, it seems that other configuration are required to make it work properly (sessionExpiry, cleanStart, ...). They appear to be defined at connection level, ex:

```java
 rxClient                         
     .connect(MqttConnect.DEFAULT.extend().sessionExpiryInterval(100000).cleanStart(false).build())
```

Note: I didn't test it. Just for note the point for a later independent task.